### PR TITLE
completed basic training :)

### PR DESCRIPTION
**Pull request by MrPalRobot**
🍡

### CODE REVIEW

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+
+#include "Person.h"
+
+
+Person::Person(){
```
Comment (DatLam25) [inline Recruit-Training/Basic-Recruit-Training/person/Person.cpp:9]: Our code formatting will have the curly brace on a line. Example:
```
Person::Person()
{
  age_ = new int(0);
}
```

Comment (DatLam25) [reply Recruit-Training/Basic-Recruit-Training/person/Person.cpp:9]: Remember to change the rest of the file.

Comment (bryce-krispies) [reply Recruit-Training/Basic-Recruit-Training/person/Person.cpp:9]: @DatLam25 We should include this in our [style guide.](https://github.com/UCSolarCarTeam/Development-Resources/tree/master/Development-Guide/Style-Guide)

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+Person::~Person(){
+    delete age_;
+}
+
+Person::Person(const std::string& name, int age){
+    name_=name;
+    age_ = new int (age);
+    *age_ = age;
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/Person.cpp:20]: This line is unnecessary - ln 19 already sets *age_ to the age

Comment (bryce-krispies) [reply Recruit-Training/Basic-Recruit-Training/person/Person.cpp:20]: Though as a cool tip, you can do this function like this:
`Person::Person(const std::string& name, int age): name_(name), age_(new int(age)){}`

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+
+
+#include "Person.h"
+
+
+Person::Person(){
+  age_ = new int (0);
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/Person.cpp:10]: As a cool tip you can change this to (through the use of _initialization lists_):
`Person::Person(): age_(new int){}`

File: Recruit-Training/Basic-Recruit-Training/person/Person.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+int Person::combinedAge(Person** personArray, int size){
+    int totalAge = 0;
+    for(int i = 0; i < size; i ++) {
+        totalAge += *(personArray[i]->age_);
+    }
+    return totalAge;
+}
+
+void Person::birthday(Person& x){
+    *(x.age_) =   *(x.age_) + 1 ;
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/Person.cpp:53]: Minor: You can do
`(*x.age_)++;`

File: Recruit-Training/Basic-Recruit-Training/person/main.cpp [hunk 5]
```diff
@@ -18,16 +18,25 @@ void compareAnswers(const std::string myAnswer, const std::string correctAnswer)
...
 }
 // Swaps a & b.
-void mySwap(int& a, int& b)
+void mySwap(int& a, int& b) // pass by reference (learned this with Dillon)
 {
+    int start = a;
+    a = b;
+    b = start;
 }
+/*void mySwap1(int a, int b) // pass by value (learned this with Dillon)
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/main.cpp:27]: Try to remove unneeded commented out code so we can reduce the file size

File: Recruit-Training/Basic-Recruit-Training/person/main.cpp [hunk 6]
```diff
@@ -18,16 +18,25 @@ void compareAnswers(const std::string myAnswer, const std::string correctAnswer)
...
+{
+    a = 4673;
+}*/
 // Increment the int that x is pointing to by one.
 void myIncrement(int* x)
 {
+    *x += 1;
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/main.cpp:34]: Minor: Try
`(*x)++`

File: Recruit-Training/Basic-Recruit-Training/person/main.cpp [hunk 7]
```diff
@@ -58,15 +67,23 @@ void myIncrement(int* x)
...
 
     // Pass a & b into mySwap here
+    mySwap(a, b);
+    std::cout <<a<< std::endl;
+    std::cout <<b<< std::endl;
+    
+        
     // Pass in a to myIncrement here
+    int* t = &a;
+    myIncrement(t);
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/main.cpp:86]: Try `myIncrement(&a)` instead

File: Recruit-Training/Basic-Recruit-Training/person/main.cpp [hunk 8]
```diff
@@ -77,9 +94,13 @@ int main()
 
     compareAnswers(swapIncrementAnswer.str(), correctSwapIncrementAnswer);
     // Pass a and b into mySwap
+    mySwap(a, b);
     // Pass in a to myIncrement
+    int *p = &a;
+    myIncrement(p);
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/person/main.cpp:100]: Try `myIncrement(&a)` instead

File: Recruit-Training/Basic-Recruit-Training/pointers/main.cpp [hunk 9]
```diff
@@ -19,31 +20,43 @@ int main()
...
 
     //start writing mySwap here
+    mySwap (p, &y);
+    cout << *p << endl;
+    cout << y << endl;
     
     return 0;
 }
 
+void mySwap(int *k, int *b)
```
Comment (bryce-krispies) [inline Recruit-Training/Basic-Recruit-Training/pointers/main.cpp:51]: Well done!

**Review by bryce-krispies — APPROVED** [review]
Overall, looks good! I left some comments/tips on how to improve your code.

**Pull request closed without merging**